### PR TITLE
Stable ID3 & ID4 Development Branch Merge to Main 

### PR DESCRIPTION
**Pull request by lecegues**
### Description
- These features are from ID3 & ID4. Due to instability, we initially did not merge it to develop. Now, after finishing up testing, reworking, and fixing bugs, we have reached a stable development branch to merge into main.

### Updates
- **Admin Login**: Developed and tests created
- **Sites**: Can create sites (and can create Site dependencies (e.g. region))
- **Projectile Points**: Can create projectile points and dependencies. Can view, but the UI is not finished.
- **Aggregate Statistics**: Can generate statistics; need to create more predefined charts
- **Testing**: End-to-end tests, smoke tests added in separate folder. Test scripts created for unit/integration tests.
- **Middleware**: Created middleware to sanitize & validate backend inputs & authenticate admins
- **CI Pipeline Workflow**: All tests run in the same job and generate test reports.

### CODE REVIEW

File: backend/config/initialize.js
Comment (DanielleDPowell) [inline backend/config/initialize.js:1]: More documentation needed

File: backend/helperFiles/artifactTypesHelper.js
Comment (DanielleDPowell) [inline backend/helperFiles/artifactTypesHelper.js:1]: - commented out code needs to be removed
- better documentation needed

File: backend/helperFiles/baseShapesHelper.js
Comment (DanielleDPowell) [inline backend/helperFiles/baseShapesHelper.js:1]: - commented out code needs to be removed
- better documentation needed

File: backend/helperFiles/bladeShapesHelper.js
Comment (DanielleDPowell) [inline backend/helperFiles/bladeShapesHelper.js:1]: - commented out code needs to be removed
- better documentation needed

File: backend/helpers/register.js
Comment (DanielleDPowell) [inline backend/helpers/register.js:1]: @lecegues Please review

File: backend/routes/aggregateStatisticsGenerators.js
Comment (DanielleDPowell) [inline backend/routes/aggregateStatisticsGenerators.js:1]: - more documentation needed

File: backend/routes/sites.js
Comment (DanielleDPowell) [inline backend/routes/sites.js:1]: for POST
- more documentation needed

File: backend/helperFiles/bladeShapesHelper.js
Comment (DanielleDPowell) [inline backend/helperFiles/bladeShapesHelper.js:1]: - more documentation needed
- remove commented out code

**Comment by lecegues** [comment]
Documentation issues will be relayed to the developer working on said feature and will be included in the next release.

**Review by DanielleDPowell — APPROVED** [review]

**Review by funmise — APPROVED** [review]
Looks good. A few things have already been noted for improvement this ID

**Pull request merged**